### ISSUE TITLE
Don't understand the "ExtendMask" Layer at the end of channel_wise_lstms in Deep Supervision

### ISSUE THREAD

**Issue by vpellegrain** [YerevaNN/mimic3-benchmarks#90]
Hi, 

I'm trying to reproduce your baselines, and I don't understand the line : 

`y = ExtendMask()([y, M])  # this way we extend mask of y to M`

I don't understand what it changes to y, knowing that M is just an initialized Input Layer. 
I'm not very comfortable with Keras / TF, mainly working with pytorch.

Thanks in advance for any insight !

**Comment by hrayrhar** [comment]
Hi, 

If I recall it correctly, when doing deep supervision in length of stay or decompensation prediction, we need to predict labels at different time points. These time points are specified by `M`. The network's predictions are `y`. That `ExtendMask` layer just tells Keras to ignore `y` in all time steps where `M` is 0. This simplifies the code as later when we the compute loss function between `y` and its target, we don't need to write code to ignore locations that were not in `M`. Keras will do it automatically by looking at the mask of `y`.
